server/game/cards/attachments/07/summer.js

```diff
@@ -0,0 +1,58 @@
+const DrawCard = require('../../../drawcard.js');
+
+class Summer extends DrawCard {
+
+    setupCardAbilities(ability) {
+        this.action({
+            title: 'Have parent participate',
+            condition: () => this.game.currentChallenge && this.game.currentChallenge.challengeType === 'military',
+            cost: ability.costs.kneelParent(),
+            limit: ability.limit.perChallenge(1),
+            handler: () => {
+                if(this.game.currentChallenge.attackingPlayer === this.controller) {
+                    this.game.currentChallenge.addAttacker(this.parent, false);
+                } else {
+                    this.game.currentChallenge.addDefender(this.parent, false);
+                }
+                
+                this.game.addMessage('{0} uses {1} and kneels {2} to have {2} participate on their side',
+                                      this.controller, this, this.parent);
+            }
+        }),
+
+        this.action({
+            title: 'Attach to a different character',
+            cost: ability.costs.payGold(1),
+            limit: ability.limit.perPhase(1),
+            target: {
+                activePromptTitle: 'Select a character',
+                cardCondition: card => (
+                    card.location === 'play area' &&
+                    card !== this.parent &&
+                    this.canAttach(this.controller, card))
+            },
+            handler: context => {
+                this.controller.removeCardFromPile(this);
+                this.parent = context.target;
+                this.moveTo('play area');
+                context.target.attachments.push(this);
```

Comment:
I think a better solution would be to modify `Player.attach` to just not raise 'onCardEntersPlay' if the card is already in play, rather than replicate the code here.